napalm/base/helpers.py

```diff
@@ -255,3 +256,47 @@ def as_number(as_number_val):
         return (int(big) << 16) + int(little)
     else:
         return int(as_number_str)
+
+
+def canonical_interface(interface, device_os, change, short=False):
+    '''
+    Function to retun interface canonical name
+    This puposely does not use regex, or first X characters, to ensure
+    there is no chance for false positives. As an example, Po = PortChannel, and
+    PO = POS. With either character or regex, that would produce a false positive.
+    '''
+    if not change:
+        return interface
+
+    if device_os == "cisco_nxos_ssh":
```

Comment:
I'd get rid of the `device_os` tbh. I don't see much value in doing this check. What we could do instead is replace the `device_os` with a `fail_on_miss` bool that does something like:

```
try:
    return reverse_mapping[long_int] + str(interface_number)
except KeyError:
    if fail_on_miss:
          raise
    return long_int+str(interface_number)
```

Reply:
I am super cautious in making sure I know I am not messing up something here. I'm fine either way, and in reality this only get's turned on if implemented in said os driver, so maybe not that big of an issue. 

I would like to keep the transposing of `cisco_nxos_ssh` -> `cisco_nxos` in case there needs to be a `base_interfaces = base_interfaces.update(interface_map.get(device_os))` on the nxos.